flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/SortKeySerializer.java

```diff
@@ -325,9 +370,13 @@ public TypeSerializerSchemaCompatibility<SortKey> resolveSchemaCompatibility(
         return TypeSerializerSchemaCompatibility.incompatible();
       }
 
-      // Sort order should be identical
       SortKeySerializerSnapshot newSnapshot =
           (SortKeySerializerSnapshot) newSerializer.snapshotConfiguration();
+      if (newSnapshot.getCurrentVersion() == 1 && this.getCurrentVersion() == 2) {
```

Comment:
In version 18, the parameter for resolveSchemaCompatibility is TypeSerializer, and it needs to be obtained through SortKeySerializerSnapshot.snapshotConfiguration() to get the SortKeySerializerSnapshot.